src/Storage/DBFile.php

```diff
@@ -506,7 +506,11 @@ public function validate(ValidationResult $result, $filename = null)
             return true;
         }
 
-        $message = _t('SilverStripe\\Assets\\File.INVALIDEXTENSION_SHORT', 'Extension is not allowed');
+        $message = _t(
+            'SilverStripe\\Assets\\File.INVALIDEXTENSION_SHORT',
+            'Extension \'{extension}\' is not allowed',
```

Comment:
I only have one concern with this in that if anyone does a patch release on 4.0, 4.1 or 4.2 and updated translations, we'll get this as a string literal pulled into new projects. I think the safer option would be to create a new translation key with this new value in it

Reply:
Cool, needs a text collector update too